From cc601c173d6bdeaf6fcc2de476db4ee7ef99607a Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 24 Oct 2016 20:49:48 +0200
Subject: [PATCH] core: drop -.slice from shipped units

Since this unit is synthesized anyway there's no point in actually shipping it
on disk. This also has the benefit that "cd /usr/lib/systemd/system ; ls *"
won't be confused by the leading dash of the file name anymore.

(cherry picked from commit 828d92acbc8e6f536419cfaee10d6b5c7b0d7f82)

Fixed up context to apply to v219.

Signed-off-by: Markus Boehme <markubo@amazon.com>
---
 Makefile.am    | 12 ------------
 units/x-.slice | 12 ------------
 2 files changed, 24 deletions(-)
 delete mode 100644 units/x-.slice

diff --git a/Makefile.am b/Makefile.am
index 648f54b957..13ac6fbb0c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -494,7 +494,6 @@ dist_systemunit_DATA = \
 	units/swap.target \
 	units/slices.target \
 	units/system.slice \
-	units/x-.slice \
 	units/systemd-initctl.socket \
 	units/systemd-shutdownd.socket \
 	units/syslog.socket \
@@ -629,17 +628,6 @@ EXTRA_DIST += \
 	units/rc-local.service.in \
 	units/halt-local.service.in
 
-# automake is broken and can't handle files with a dash in front
-# http://debbugs.gnu.org/cgi/bugreport.cgi?bug=14728#8
-units-install-hook:
-	mv $(DESTDIR)$(systemunitdir)/x-.slice $(DESTDIR)/$(systemunitdir)/-.slice
-
-units-uninstall-hook:
-	rm -f $(DESTDIR)/$(systemunitdir)/-.slice
-
-INSTALL_DATA_HOOKS += units-install-hook
-UNINSTALL_DATA_HOOKS += units-uninstall-hook
-
 dist_doc_DATA = \
 	README \
 	NEWS \
diff --git a/units/x-.slice b/units/x-.slice
deleted file mode 100644
index ac82c35874..0000000000
--- a/units/x-.slice
+++ /dev/null
@@ -1,12 +0,0 @@
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[Unit]
-Description=Root Slice
-Documentation=man:systemd.special(7)
-DefaultDependencies=no
-Before=slices.target
-- 
2.39.1

